c/certificateService.c

```diff
@@ -118,36 +119,83 @@ static void respondWithBadRequest(HttpResponse *response) {
 
     jsonStart(p);
     {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
+      jsonAddString(p, "error", value);
     }
     jsonEnd(p);
 
     finishResponse(response);
 }
 
+
 static int serveMappingService(HttpService *service, HttpResponse *response)
 {
   HttpRequest *request = response->request;
 
-  if (!strcmp(request->method, methodPOST))
-  {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
+  if (!strcmp(request->method, methodPOST)) {
+    RUsermapParamList userMapStructure = {0};
 
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
+    int urlLength = strlen(request->uri);
+    if (urlLength < 0 || urlLength > 64) {
+      respondWithBadRequest(response, "URL length is not in range from 0 to 64 bytes.");
       return 0;
     }
+    char translatedURL[urlLength + 1];
```

Comment:
Bug: this is still left unterminated.

You can do
```c
char translatedURL[urlLength + 1] = {0};
```

Or use `strcpy` instead of `memcpy`.